### DISCUSSION TITLE
Where are the breakpoints stored?

### DISCUSSION THREAD

**Discussion by raulperez** [eclipse-platform/eclipse.platform#797] [Q&A]
Hi

Is it possible to get access to the list of breakpoints stored in the metadata? Is there a way to see that list? Thanks in advance.

**Comment by fedejeanne** [comment]
Hola Raul,
I see there's no answer yet. Maybe JDT people can help? --> https://github.com/eclipse-jdt/eclipse.jdt.debug/discussions

**Comment by mickaelistria** [comment]
The documentation of IBreakpoint says "Each breakpoint has an associated marker which stores and persists all attributes associated with a breakpoint." and other interesting things.
I'm not sure whether it's true for all cases (for instance, if I setup a class initialization breakpoint, how is it stored?), so it's possible that JDT does something else.

**Comment by iloveeclipse** [comment]
They just markers with attributes.

**Comment by fedejeanne** [comment]
That hit the nail in the head, thank you both.

They are stored in a `.markers`-file in the metadata of `org.eclipse.core.resources`, inside a subfolder with the name of the project where they reside _i.e._ `<WORKSPACE>\.metadata\.plugins\org.eclipse.core.resources\.projects\<PROJECT>\.markers`

If you for example add 3 breakpoints in a class inside the project `org.eclipse.search.core` and one breakpoint in a class inside the project `org.eclipse.jdt.apt.pluggable.tests` and then search those files for the string `Line breakpoint` then you will find 3 hits in one file and 1 hit in another file:

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/cbd61260-187b-48c3-a898-0be8ca731cf8)

Does this answer your question?

**Comment by Bananeweizen** [comment]
File > Export > Breakpoints would be the far more simple solution probably.
